awx/main/models/inventory.py

```diff
@@ -1047,7 +1029,7 @@ def cloud_credential_validation(source, cred):
         # Allow an EC2 source to omit the credential.  If Tower is running on
         # an EC2 instance with an IAM Role assigned, boto will use credentials
         # from the instance metadata instead of those explicitly provided.
-        elif source in CLOUD_PROVIDERS and source not in ['ec2', 'openshift_virtualization']:
+        elif source in discover_available_cloud_provider_plugin_names and source not in ['ec2', 'openshift_virtualization']:
```

Comment:
This is a function, not an iterable:
```suggestion
        elif source in discover_available_cloud_provider_plugin_names() and source not in ['ec2', 'openshift_virtualization']:
```